httpconnect/httpconnect_test.go

```diff
@@ -1,55 +1,55 @@
 package httpconnect
 
 import (
+	"fmt"
 	"net"
 	"net/http"
 	"net/http/httptest"
 	"net/url"
+	"strconv"
 	"testing"
+	"time"
 
 	"github.com/getlantern/http-proxy/filters"
 	"github.com/stretchr/testify/assert"
 )
 
 func TestFilterTunnelPorts(t *testing.T) {
+	origin := httptest.NewUnstartedServer(http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+		w.WriteHeader(200)
+		w.Write([]byte("hi"))
+	}))
+	origin.StartTLS()
+	defer origin.Close()
+	ou, _ := url.Parse(origin.URL)
+	_, _port, _ := net.SplitHostPort(ou.Host)
+	port, _ := strconv.Atoi(_port)
+
 	server := httptest.NewServer(filters.Join(
-		New(&Options{AllowedPorts: []int{443, 8080}}),
+		New(&Options{AllowedPorts: []int{port, 443}, IdleTimeout: 30 * time.Second}),
 		filters.Adapt(http.NotFoundHandler())))
 	defer server.Close()
 	u, _ := url.Parse(server.URL)
 	client := http.Client{Transport: &http.Transport{
-		Dial: func(network, addr string) (net.Conn, error) {
-			return net.Dial("tcp", u.Host)
+		Proxy: func(req *http.Request) (*url.URL, error) {
+			return u, nil
 		},
+		DisableKeepAlives: true,
 	}}
 
-	req, _ := http.NewRequest("CONNECT", "http://site.com", nil)
+	req, _ := http.NewRequest("GET", "https://site.com:abc", nil)
```

Comment:
This means that GET requests get treated like a CONNECT?
The code for checking the method as CONNECT hasn't changed, why remove the tests for CONNECT otherwise?